cache/remotecache/v1/chains.go

```diff
@@ -126,6 +139,30 @@ func (c *item) LinkFrom(rec solver.CacheExporterRecord, index int, selector stri
 	}
 
 	c.links[index][link{src: src, selector: selector}] = struct{}{}
+	src.backlinks[c] = struct{}{}
+}
+
+func (c *item) validate() {
+	for _, m := range c.links {
+		if len(m) == 0 {
+			c.invalid = true
```

Comment:
@tonistiigi So the only case where `c.invalid == false` is if `len(m) > 0` ? I'm not sure I understand everything, but am just curious why we check for backlinks only if `len(m) == 0` ?

Reply:
`len(m) == 0` means links are incomplete and invalid